### PR TITLE
remove clients field from adb_context_s and related stuff

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
to avoid access the freed memory after adb_destroy_client

### CODE REVIEW

File: adb.h [hunk 1]
```diff
@@ -157,7 +157,6 @@ typedef struct adb_client_s {
 } adb_client_t;
 
 typedef struct adb_context_s {
-    adb_client_t *clients;
```
Comment (spiriou) [inline adb.h:155]: The idea behind tracking all the clients is it's then possible to terminate properly the server, releasing all the resources. This is unfortunately not implemented yet so there is an `adb_register_client()` function without `adb_UNregister_client()`.

It's fine for now to remove it as the feature is not implemented. In this case `client->next` becomes useless too.

Comment (xiaoxiang781216) [reply adb.h:155]: @spiriou rebase to fix the conflict.

Comment (spiriou) [reply adb.h:155]: Can you remove the `next` field too in `adb.h:struct adb_client`?
```
142 typedef struct adb_client_s {
143    struct adb_client_s *next;
```

Comment (xiaoxiang781216) [reply adb.h:155]: Done.

**Pull request merged**